spec/models/miq_queue_spec.rb

```diff
@@ -756,6 +756,82 @@ def self.some_method(single_arg)
     end
   end
 
+  describe ".broadcast" do
+    let(:queue_params) do
+      {
+        :class_name  => 'MyClass',
+        :method_name => 'method1',
+        :args        => [1, 2]
+      }
+    end
+
+    def queue_items
+      MiqQueue.where(queue_params.slice(:class_name, :method_name))
+    end
+
+    context "with no servers" do
+      it "nothing is created" do
+        MiqQueue.broadcast(queue_params)
+
+        expect(queue_items.count).to eq(0)
+      end
+    end
+
+    context "with a single server" do
+      it "creates a queue item for the server" do
+        EvmSpecHelper.create_guid_miq_server_zone
+        MiqQueue.broadcast(queue_params)
+
+        expect(queue_items.count).to eq(1)
+        expect(MiqQueue.get).to have_attributes(queue_params)
+      end
+    end
+
+    context "with servers in two different zones" do
+      let(:other_zone) { FactoryBot.create(:zone) }
+
+      # NOTE: `.create_list` doesn't work with `:guid`
+      let(:other_servers) do
+        [
+          FactoryBot.create(:miq_server, :guid => SecureRandom.uuid, :zone => other_zone),
+          FactoryBot.create(:miq_server, :guid => SecureRandom.uuid, :zone => other_zone)
+        ]
+      end
+
+      before do
+        EvmSpecHelper.create_guid_miq_server_zone
+        other_servers
+      end
+
+      it "creates a queue item for the server" do
+        MiqQueue.broadcast(queue_params)
+
+        expect(queue_items.count).to eq(3)
+
+        (other_servers + [MiqServer.my_server]).each do |server|
+          allow(MiqServer).to receive(:my_guid).and_return(server.guid)
+          MiqServer.my_server_clear_cache
+
+          expect(MiqQueue.get).to have_attributes(queue_params)
+        end
+      end
+
+      it "is filterable by :zone" do
+        MiqQueue.broadcast(queue_params.merge(:zone => other_zone))
+
+        expect(queue_items.count).to eq(2)
+        expect(MiqQueue.get).to eq(nil)
+
+        other_servers.each do |server|
+          allow(MiqServer).to receive(:my_guid).and_return(server.guid)
+          MiqServer.my_server_clear_cache
```

Comment:
These lines are a bit nasty. I think we can do this without stubbing this out.

I would rather query `MiqQueue` directly than use the `.get` method. Then you don't have to stub the current server out.